File: sdks/java/src/main/java/dev/atap/sdk/exception/ATAPAuthException.java
package dev.atap.sdk.exception;

import dev.atap.sdk.model.ProblemDetail;

/**
 * Authentication or authorization error (401/403).
 */
public class ATAPAuthException extends ATAPException {

    private final ProblemDetail problem;

    public ATAPAuthException(String message, int statusCode, ProblemDetail problem) {
        super(message, statusCode);
        this.problem = problem;
    }

    public ATAPAuthException(String message, int statusCode) {
        this(message, statusCode, null);
    }

    public ProblemDetail getProblem() {
        return problem;
    }
}

File: sdks/php/src/Exception/ATAPAuthException.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Exception;

use Atap\Sdk\Model\ProblemDetail;

/**
 * Authentication or authorization error (401/403).
 */
class ATAPAuthException extends ATAPException
{
    public function __construct(
        string $message = 'Authentication failed',
        int $statusCode = 401,
        public readonly ?ProblemDetail $problem = null,
        ?\Throwable $previous = null,
    ) {
        parent::__construct($message, $statusCode, $previous);
    }
}
